model.py

```diff
@@ -8662,16 +8662,27 @@ def metadata_identifier(self):
         In the metadata wrangler, this identifier is used as the unique
         name of the collection.
         """
-        account_id_column = self.UNIQUE_IDENTIFIER_BY_PROTOCOL.get(self.protocol)
-        account_id = getattr(self, account_id_column.name)
+        def get_account_id(collection):
+            account_id_column = self.UNIQUE_IDENTIFIER_BY_PROTOCOL.get(collection.protocol)
+            return getattr(collection, account_id_column.name)
+
+        collection = self
+        account_ids = list()
+        while collection:
+            account_id = get_account_id(collection)
```

Comment:
I think you could simplify this quite a bit by defining `account_ids` as a recursive method that got `parent.account_ids` as a list and stuck the final account ID on the end.